content/tutorials/quickstart/py-create-conditional-escrows.md

```diff
@@ -0,0 +1,625 @@
+
+## getConditionAndFulfillment.js
+
+To generate a condition/fulfillment pair, use Node.js to run the `getConditionAndFulfillment.js` script.
+
+```javascript
+function getConditionAndFulfillment() {
+```
+
+Instantiate the `five-bells-condition` and `crypto` libraries.
+
+```javascript
+  const cc = require('five-bells-condition')
+  const crypto = require('crypto')
+```
+
+Create a random 32-byte seed string.
+
+```javascript
+  const preimageData = crypto.randomBytes(32)
+```
+
+Create a fulfillment object.
+
+```javascript
+  const fulfillment = new cc.PreimageSha256()
+```
+
+Generate a fulfillment code.
+
+```javascript
+  fulfillment.setPreimage(preimageData)
+```
+
+Generate the condition value based on the fulfillment value.
+
+```javascript
+  const condition = fulfillment.getConditionBinary().toString('hex').toUpperCase()
+```
+
+Return the condition.
+
+```javascript
+  console.log('Condition:', condition)
+```
+
+Convert the fulfillment code to a hexadecimal string.
+
+```javascript
+  const fulfillment_hex = fulfillment.serializeBinary().toString('hex').toUpperCase()
+```
+
+Return the fulfillment code. Keep it secret until you want to finish the escrow.
+
+```javascript
+  console.log('Fulfillment:', fulfillment_hex)
+}
+getConditionAndFulfillment()
+```
+
+## mod9.py
+
+Import dependencies.
+
+```python
+import xrpl
+from xrpl.clients import JsonRpcClient
+from xrpl.wallet import Wallet
+from datetime import datetime
+from xrpl.models.transactions import EscrowCreate, EscrowFinish
+```
+
+Create a global variable pointing to Testnet.
+
+```python
+testnet_url = "https://s.altnet.rippletest.net:51234"
+```
+
+### add_seconds
+
+Create a date in the Ripple epoch, adding the specified number of seconds.
+
+```python
+def add_seconds(numOfSeconds):
+```
+
+Create a new_date variable.
+
+```python
+    new_date = datetime.now()
+```
+
+Convert the date to a Ripple time object.
+
+```python
+    if new_date != '':
+        new_date = xrpl.utils.datetime_to_ripple_time(new_date)
+```
+
+Add the requested seconds to the Ripple formatted date variable.
+
+```python
+        new_date = new_date + int(numOfSeconds)
+```
+
+Return the result.
+
+```python
+    return new_date
+```
+
+### create_conditional_escrow
+
+You create conditional escrows using the same **EscrowCreate** model you used for a time-based escrow, but instead of a finish time, you provide a condition that must be met to complete the transaction.
+
+Pass the _seed_ for the sending account, the _amount_ to hold in escrow, the _destination_ account to receive the escrow funds, the number of seconds until the escrow will _cancel_, and a _condition_ value that will be matched with a _fulfillment_ value to complete the escrow.
+
+```python
+def create_conditional_escrow(seed, amount, destination, cancel, condition):
+```
+
+Instantiate a wallet and connect to Testnet.
+
+```python
```

Comment:
This section about instantiating wallets should be in a different section.  It interupts the flow as both your methods `create_conditional_escrow` and `finish_conditional_escrow` do the same thing.  It would be better to just pass in the client and wallet.

While it does add additional params it remove this repetitive section.

I could be wrong.

Reply:
For these examples, it seems that would make sense. But keep in mind that the create and finish escrows are executed on different machines after days or weeks. I think the current approach is more in line with practical usage.